### ISSUE TITLE
How to get the speaker embedding duration training of base model?

### ISSUE THREAD

**Issue by JohnHerry** [FunAudioLLM/CosyVoice#246]
Hi, I read in the paper that a x-vector encoder was pretrained to extract the speaker vector v.  
there are two parts where the vector is used:  the LLM model and the flow-matching model.  
then how did the speaker embedding generated for use in the two model training? 
 maybe 1)  During training, I get the x-vector with the current target speech mel as input.    `v = x_vec(speech_target)`  
 maybe 2)  During training, I get the x-vector with a random speech sample from the same speaker of current target speech as input.   `v = x_vec(speech_x)   WHEN speaker_of(speech_x) == speaker_of(speech_target)`  
 maybe 3)  During preprocessing, I get the average of all speechs' x-vector embedding under the same speaker to get a  static， **fixed** speaker embeding  for **all samples of the same speaker**. and then that value will be used during training and inference to label the speaker:   `v = average( x_vec(all_speechs_of_current_speaker))`    
which one is the truth?

**Comment by aluminumbox** [comment]
during training, we use xvec from the current speech, but actually you can also use random xvec for the spk, our code tried to do it by concate all utt embedding into spk embedding, but it takes too much disk space. Random xvec during training should improve zero shot inference performance

**Comment by JohnHerry** [comment]
Thanks,  We had tried TorToiseTTS with speaker condition from the current speech, but the result is terrible, while the case to use random uttrance from the same speaker get accessable result. may be our dataset is too small,  we had less then 10k hours dataset.
